provisioner/src/lib.rs

```diff
@@ -236,6 +258,294 @@ impl MyProvisioner {
+        }
+
+        let key = self
+            .iam_client
+            .create_access_key()
+            .user_name(self.get_iam_identity_user_name(prefix).await)
+            .send()
+            .await
+            .map_err(Error::CreateAccessKey)?;
+        let access_key = key
+            .access_key()
+            .ok_or_else(|| Error::GetAccessKey("empty access key".to_string()))?;
+
+        let access_key_id = access_key
+            .access_key_id
+            .as_ref()
+            .ok_or_else(|| Error::GetAccessKeyId("empty access key id".to_string()))?
+            .to_string();
+        let secret_access_key = access_key
+            .secret_access_key
+            .as_ref()
+            .ok_or_else(|| Error::GetSecretAccessKey("empty access key secret".to_string()))?
+            .to_string();
+
+        self.save_access_key(prefix, &access_key_id, &secret_access_key)
+            .await
+            .map_err(Error::GetIAMIdentityKeys)?;
+
+        Ok((access_key_id, secret_access_key))
+    }
+
+    async fn delete_access_key(&self, prefix: &str) -> Result<(), Error> {
+        let (access_key_id, _secret_access_key) = self.get_iam_identity_keys(prefix).await?;
+
+        self.iam_client
+            .delete_access_key()
+            .user_name(self.get_iam_identity_user_name(prefix).await)
+            .access_key_id(access_key_id)
+            .send()
+            .await
+            .map_err(Error::DeleteAccessKey)?;
+
+        self.delete_saved_access_key(prefix).await?;
+
+        Ok(())
+    }
+
+    async fn delete_iam_identity(&self, prefix: &str) -> Result<DeleteUserOutput, Error> {
+        let user = self
+            .iam_client
+            .delete_user()
+            .user_name(self.get_iam_identity_user_name(prefix).await)
+            .send()
+            .await
+            .map_err(Error::DeleteIAMUser)?;
+        Ok(user)
+    }
+
+    async fn attach_user_policy(&self, prefix: &str) -> Result<(), Error> {
+        self.iam_client
+            .attach_user_policy()
+            .user_name(self.get_iam_identity_user_name(prefix).await)
+            .policy_arn(self.get_policy_arn(prefix).await?)
+            .send()
+            .await
+            .map_err(Error::AttachUserPolicy)?;
+        Ok(())
+    }
+
+    async fn detach_user_policy(&self, prefix: &str) -> Result<(), Error> {
+        self.iam_client
+            .detach_user_policy()
+            .user_name(self.get_iam_identity_user_name(prefix).await)
+            .policy_arn(self.get_policy_arn(prefix).await?)
+            .send()
+            .await
+            .map_err(Error::DetachUserPolicy)?;
+        Ok(())
+    }
+
+    pub async fn request_dynamodb(&self, project_name: &str) -> Result<DynamoDbResponse, Error> {
+        let prefix = self.get_prefix(project_name).await;
+
+        self.create_dynamodb_policy(&prefix).await?;
+
+        self.create_iam_identity(&prefix).await?;
+
+        self.attach_user_policy(&prefix).await?;
```

Comment:
I'm wondering, since all these methods just need the prefix and the IAM client, maybe all this logic can move to a separate struct. This struct would have a small lifetime and would be constructed here?

I'm mostly trying to find a way to make this current struct's `impl` simpler and move the new common logic to one place.